components/ForgotPassword.tsx

```diff
@@ -0,0 +1,119 @@
+"use client";
+
+import Image from "next/image";
+import { useState } from "react";
+
+// Logo Import
+import logo from "../public/Cacart_logo.png";
+
+// Image Import
+import forgotPasswordImage from "../public/forgotPassword.png";
+import dottedShapeImage from "../public/Dotted Shape.png";
+
+export default function ForgotPassword() {
+  const [userData, setUserData] = useState({
+    email: "",
+  });
+  const [emailError, setEmailError] = useState("");
+  const [emailValid, setEmailValid] = useState(false);
+
+  // Value update handler
+  const valueUpdateHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setUserData((prevData) => ({
+      ...prevData,
+      [name]: value,
+    }));
+  };
+
+  const isEmailValid = (email: any) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```

Comment:
How about we separate the regex expression into a file and contains all the regex expressions we would use in the app? We would need password regex, and might need some other regex expressions along the way as well. 

Also, since regex is cryptic to human 😅,  how about we add comments explaining what it does? (ex, looks for '@' symbol in string in xx location)

Reply:
DONE, created regexPatterns.tsx